packages/aws-cdk-lib/aws-ecs/lib/log-drivers/aws-log-driver.ts

```diff
@@ -106,6 +117,10 @@ export class AwsLogDriver extends LogDriver {
     if (props.logGroup && props.logRetention) {
       throw new Error('Cannot specify both `logGroup` and `logRetentionDays`.');
     }
+
+    if (props.maxBufferSize !== undefined && props.mode !== AwsLogDriverMode.NON_BLOCKING) {
```

Comment:
No that we're using `Size` as the type we can just use:

`if (props.maxBufferSize && props.mode !== AwsLogDriverMode.NON_BLOCKING`